### ISSUE TITLE
Enum values are not checked by IDE1006 (naming rules)

### ISSUE THREAD

**Issue by detzt** [dotnet/roslyn#72574]
**Version Used**: Latest

**Steps to Reproduce**:

1. Setup an .editorconfig that tries to style enum values, e.g.
```csharp
dotnet_naming_symbols.all_members.applicable_kinds = *
dotnet_naming_rule.members_should_be_pascal_case.symbols = *
dotnet_naming_style.pascal_case.capitalization = pascal_case
dotnet_naming_rule.members_should_be_pascal_case.style = pascal_case
dotnet_naming_rule.members_should_be_pascal_case.severity = warning
```
2. Define an enum that violates this naming style, e.g.
```csharp
public enum shouldBeUpperCaseEnum { shouldBeUppCaseEnumValue }
```

**Diagnostic Id**: IDE1006

**Expected Behavior**:
IDE1006 is triggered for `shouldBeUpperCaseEnum` and `shouldBeUppCaseEnumValue`.

**Actual Behavior**:
IDE1006 is only triggered for `shouldBeUpperCaseEnum`.
Nothing is shown for `shouldBeUppCaseEnumValue` regardless of how .editorconfig is set up.

**Comment by sharwell** [comment]
Duplicate of #24209

**Comment by detzt** [comment]
I see the similarity between the two issues and if the other issue gets fixed, then this issue will probably be fixed as well. However, I understand the other issue as a feature request to have different style definitions for enum members and const fields.
In contrast, this issue is a bug report that the name of enum members is not checked at all.

No matter how I configure the naming rules for all symbols, const fields, or enums, and no matter how I name an enum member, it never triggers an IDE1006 Naming rule violation. All other kinds are triggering IDE1006 when not named properly.

@sharwell Does any test setup on your machine show a naming rule violation for any enum member?

**Comment by sharwell** [comment]
No, there is no setting currently to apply a naming rule to an enum member.

**Comment by detzt** [comment]
Reading the comments further down on the other issue, I now see that the behavior has changed over time.
I now understand the situation and why this issue is considered a duplicate.

However, the other issue is labeled as Feature Request and given its title and description, I have a feeling it won't get prioritized properly. But that is up to you, I'll now stop bothering you :)

**Comment by CyrusNajmabadi** [comment]
@detzt THe issue is currently:

![image](https://github.com/dotnet/roslyn/assets/4564579/144df848-9903-4f1f-895c-dc672b90aabc)

Which means we're def taking community contributions here.  If this is important to you, def consider contributing to roslyn here as that will then have a much higher likelihood of hte change making it out to you in a short timeframe :)
